gtk4/src/subclass/tree_model.rs

```diff
@@ -0,0 +1,503 @@
+) {
+    let instance = &*(tree_model as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    let tree: Borrowed<TreePath> = from_glib_borrow(treeptr);
+
+    imp.row_deleted(
+        from_glib_borrow::<_, TreeModel>(tree_model).unsafe_cast_ref(),
+        &tree,
+    )
+}
+
+unsafe extern "C" fn tree_model_get_flags<T: TreeModelImpl>(
+    tree_model: *mut ffi::GtkTreeModel,
+) -> ffi::GtkTreeModelFlags {
+    let instance = &*(tree_model as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    imp.get_flags(from_glib_borrow::<_, TreeModel>(tree_model).unsafe_cast_ref())
+        .to_glib()
+}
+
+unsafe extern "C" fn tree_model_get_n_columns<T: TreeModelImpl>(
+    tree_model: *mut ffi::GtkTreeModel,
+) -> i32 {
+    let instance = &*(tree_model as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    imp.get_n_columns(from_glib_borrow::<_, TreeModel>(tree_model).unsafe_cast_ref())
+}
+
+unsafe extern "C" fn tree_model_get_column_type<T: TreeModelImpl>(
+    tree_model: *mut ffi::GtkTreeModel,
+    index: i32,
+) -> glib::ffi::GType {
+    let instance = &*(tree_model as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    imp.get_column_type(
+        from_glib_borrow::<_, TreeModel>(tree_model).unsafe_cast_ref(),
+        index,
+    )
+    .to_glib()
+}
+
+unsafe extern "C" fn tree_model_get_iter<T: TreeModelImpl>(
+    tree_model: *mut ffi::GtkTreeModel,
+    iterptr: *mut ffi::GtkTreeIter,
+    pathptr: *mut ffi::GtkTreePath,
+) -> glib::ffi::gboolean {
+    let instance = &*(tree_model as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    let path: Borrowed<TreePath> = from_glib_borrow(pathptr);
+
+    let ret = imp.get_iter(
+        from_glib_borrow::<_, TreeModel>(tree_model).unsafe_cast_ref(),
+        &path,
+    );
+    if let Some(iter) = ret {
+        *iterptr = *iter.to_glib_none().0;
+        true.to_glib()
+    } else {
+        false.to_glib()
+    }
+}
+
+unsafe extern "C" fn tree_model_get_path<T: TreeModelImpl>(
+    tree_model: *mut ffi::GtkTreeModel,
+    iterptr: *mut ffi::GtkTreeIter,
+) -> *mut ffi::GtkTreePath {
+    let instance = &*(tree_model as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    let iter: Borrowed<TreeIter> = from_glib_borrow(iterptr);
+
+    imp.get_path(
+        from_glib_borrow::<_, TreeModel>(tree_model).unsafe_cast_ref(),
+        &iter,
+    )
+    .to_glib_full() as *mut _
+}
+
+unsafe extern "C" fn tree_model_get_value<T: TreeModelImpl>(
+    tree_model: *mut ffi::GtkTreeModel,
+    iterptr: *mut ffi::GtkTreeIter,
+    index: i32,
+    valueptr: *mut glib::gobject_ffi::GValue,
+) {
+    let instance = &*(tree_model as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    let iter: Borrowed<TreeIter> = from_glib_borrow(iterptr);
+
+    let ret = imp.get_value(
+        from_glib_borrow::<_, TreeModel>(tree_model).unsafe_cast_ref(),
+        &iter,
+        index,
+    );
+    *valueptr = *ret.to_glib_none().0;
```

Comment:
This is wrong. You'd need to `mem::forget()` the value afterwards or use `ManuallyDrop` (better), otherwise the content of the value will be freed twice

Also the C annotations are completely wrong. This should be `out caller-allocates` AFAIU.